File: android/src/main/java/com/getcapacitor/community/BridgefyConfig.java
package com.getcapacitor.community;

import static com.getcapacitor.community.BridgefyHelper.makeUUID;

import java.util.UUID;
import javax.annotation.Nullable;

public class BridgefyConfig {

    @Nullable
    UUID userID;

    @Nullable
    UUID apiKey;

    @Nullable
    Boolean verboseLogging;

    @Nullable
    String propagationProfile;

    public BridgefyConfig(
        @Nullable UUID userID,
        @Nullable String apiKey,
        @Nullable Boolean verboseLogging,
        @Nullable String propagationProfile
    ) {
        this.userID = userID;

        this.setAPIKey(apiKey);
        this.setVerboseLogging(verboseLogging);
        this.setPropagationProfile(propagationProfile);
    }

    public void setAPIKey(@Nullable String apiKey) {
        this.apiKey = makeUUID(apiKey);
    }

    public void setVerboseLogging(@Nullable Boolean verboseLogging) {
        this.verboseLogging = verboseLogging;
    }

    public void setPropagationProfile(@Nullable String propagationProfile) {
        this.propagationProfile = propagationProfile;
    }

    @Nullable
    public UUID getAPIKey() {
        return apiKey;
    }

    @Nullable
    public Boolean getVerboseLogging() {
        return verboseLogging;
    }

    @Nullable
    public String getPropagationProfile() {
        return propagationProfile;
    }

    @Nullable
    public UUID getUserID() {
        return userID;
    }
}

File: ios/Sources/BridgefyPlugin/BridgefyConfig.swift
import Capacitor

public class BridgefyConfig {
    private var userID: UUID?

    private var apiKey: UUID?
    private var verboseLogging: Bool?

    private var propagationProfile: String?

    init(config: PluginConfig) {
        // self.userID = UIDevice.current.identifierForVendor

        if let apiKey = config.getString("apiKey") {
            self.setAPIKey(apiKey)
        }

        if let verboseLogging = config.getString("verboseLogging") {
            self.setVerboseLogging(verboseLogging)
        }

        if let propagationProfile = config.getString("propagationProfile") {
            self.setPropagationProfile(propagationProfile)
        }
    }

    func setAPIKey(_ apiKey: String?) {
        self.apiKey = Helper.makeUUID(apiKey)
    }
    func setVerboseLogging(_ verboseLogging: String?) {
        self.verboseLogging = Helper.makeBoolean(verboseLogging)
    }
    func setPropagationProfile(_ propagationProfile: String?) {
        self.propagationProfile = propagationProfile
    }

    func getAPIKey() -> UUID? {
        return self.apiKey
    }
    func getVerboseLogging() -> Bool? {
        return self.verboseLogging
    }
    func getPropagationProfile() -> String? {
        return self.propagationProfile
    }

    func getUserID() -> UUID? {
        return self.userID
    }
}
